### ISSUE TITLE
Func<T1?,T2?> should not warn when assigned Func<int, string>

### ISSUE THREAD

**Issue by HobbsCode** [dotnet/roslyn#63176]
**Version Used**: 
.NET 6.0.302 

**Steps to Reproduce**:

Passing a Func<int, string> to a Func<T1?, T2?> causes a warning but only when both T1 and T2 are nullable.
I don't think there should be any warning.

```
    public static void Main() 
    {        
        Func<int, string> f = i => i.ToString();
        
        Foo_1(f); 
        Foo_2(f); 
        Foo_3(f);
        Foo_4(f); // Warning!
    }
    
    public static void Foo_1<T1, T2>(Func<T1 , T2 > func){} // OK.
    public static void Foo_2<T1, T2>(Func<T1 , T2?> func){} // OK.
    public static void Foo_3<T1, T2>(Func<T1?, T2 > func){} // OK.
    public static void Foo_4<T1, T2>(Func<T1?, T2?> func){} // Both T1 and T2 nullable causes warning. 
```

[sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQAxAHYCuANhQIbAUwAEMJt9hhGAzA9gwMINCAbwaCCo0dixiJDEbImEFEjAFYAPAEsSAFwA0PHAAYAfAwBmDALwNN1s5oB0AFQgBlHVG0BzABQBKAG4lZRlQgDEICAB9HF9zILDlSJiseMSQ5KjoznTg8QjslDyGAHpShgB1aigSHwBCTIYAXyamrkMkHhQGFNj1ZxwDZywTXzUBnAZh6TNzMhIwfyFmsoqAeQBpR3buDBwujB6+rEmZsYnB6YYRgH45haWVtYYtnYLJPYPu3uzOM5uo3GGkGtxmDAei2Wq3Kr22u06Pz6KABIwuIJwYMB9wsj2hLwAQhAdAALG5TagkAAmgIY5CorEYYGoZAAzjBWQwAO41OokbyOWRtAjNIA==)

**Expected Behavior**:
No warnings.

**Actual Behavior**:
Warning.

**Comment by CyrusNajmabadi** [comment]
this is because your argument type is int.  so the inferred type for `T1` which as `T1?` is no different (since this is an unconstrained type parameter).  If you had `Func<string, string>` you would see additional warnings as `string` vs `string?` are different, including for unconstrained type parameters.
